### ISSUE TITLE
Submitting a new profile results in two web3 transaction requests

### ISSUE THREAD

**Issue by justinkalland** [Proof-Of-Humanity/proof-of-humanity-web#308]
**Describe the Bug**
As a new user when I submit a new profile MetaMask prompts me to confirm two transactions (they are byte-for-byte identical). Confirming both would end it wasting gas as one would be denied.

Two users reported this to me so I decided to check it out for myself. It caused confusion because they thought they needed to send both transactions to request registration.

**To Reproduce**
- Clear cache.
- Start application with a funded MetaMask wallet (for an address that isn't registered on PoH).
- Click "Submit Profile"
- Fill out required information.
- Click Submit
- MetaMask asks you two confirm two transactions

**Expected Behavior**
I expect the application to only submit one transaction to the web3 wallet.

**Screenshots**
<img width="378" alt="Screen Shot 2021-05-16 at 9 48 31 AM" src="https://user-images.githubusercontent.com/43973143/118390019-35f58b00-b62d-11eb-93d0-b911a0a29761.png">

Video of reproduction: https://drive.google.com/file/d/11R3jva-KwWmUOBHNT_xuKGCfRdTzTA2K/view?usp=sharing

**Device and Browser Information**
Please see video

**Comment by justinkalland** [comment]
Parsing through some newer submissions, I can see people have made the mistake of confirming both transactions:

- https://etherscan.io/address/0xe2dc6c29b31a39c416ac325bee52944677187f79
- https://etherscan.io/address/0x9f82a22b62686e77964e71ddec96d3646990404b
- https://etherscan.io/address/0xa64092c19f547f33cfafb15d5a1380dc80545d14
- https://etherscan.io/address/0x3af98916d560ccb524180597e6a35f2201ef5df3
- https://etherscan.io/address/0xf2e9ec4ba09bae56177e795c8ae7d4a9e9b16669
- https://etherscan.io/address/0x3057587f759808688bce6f0022115269295ac398

**Comment by justinkalland** [comment]
Looking through mainnet transactions again it appears some users are also clicking Submit more than once, and approving all the resulting transactions 😳

**Comment by luiandresgonzalez** [comment]
I also have this issue. The double submission also affects other transactions like fund submission button or the challenge removal button. People on telegram groups also report this issue, so it looks like it is widespread.

**Comment by justinkalland** [comment]
> I also have this issue. The double submission also affects other transactions like fund submission button or the challenge removal button. People on telegram groups also report this issue, so it looks like it is widespread.

Agreed, playing around with debugging it seems state management in the UI needs some love.

**Comment by justinkalland** [comment]
This commit is what created the current bug: https://github.com/Proof-Of-Humanity/proof-of-humanity-web/commit/c561920fede8e82458a7aa61cf2943b021ef385e#diff-5d04351f46fd31e3ab5abf320c7ada976f3ba8b63366edf5f1d1aba3f3991b66L439-L440

**Comment by michaelsproul** [comment]
I just ran into this bug (or something similar) today while registering. I was using MetaMask 10.8.0, Firefox 95.0.1 and macOS 11.6.

I initially thought I might have clicked Submit twice accidentally, but after rejecting the first two transactions and resubmitting I got prompted with two more transactions.

If you'd like me to try reproducing it on the same setup let me know what logs/debugging info to record :pray:
